controllers/events.js

```diff
@@ -0,0 +1,207 @@
+/* eslint-disable camelcase */
+const { EventAPIService } = require("../services/EventAPIService");
+const { EventTokenService } = require("../services/EventTokenService");
+const eventQuery = require("../models/events");
+const logger = require("../utils/logger");
+const { GET_ALL_EVENTS_LIMIT_MIN } = require("../constants/events");
+const { removeUnwantedProperties } = require("../utils/events");
+
+const tokenService = new EventTokenService();
+const apiService = new EventAPIService(tokenService);
+
+/**
+ * Creates a new event document in the Firestore database with the data provided in the HTTP request body.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The saved event data in JSON format.
+ * @throws {Error} If an error occurs while creating the event document.
+ */
+const createEvent = async (req, res) => {
+  const { name, description, region, userId } = req.body;
+  const payload = { name, description, region };
+  try {
+    const eventData = await apiService.post("/rooms", payload);
+    const propertiesToRemove = ["customer_id", "app_id", "recording_info", "template_id", "template", "customer"];
+    const event = removeUnwantedProperties(propertiesToRemove, eventData);
+    const eventDataFromDB = await eventQuery.createEvent({ room_id: eventData.id, created_by: userId, ...event });
+    return res.status(201).json(eventDataFromDB);
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't create event. Please try again later",
+    });
+  }
+};
+
+/**
+ * Retrieves all events that match the query parameters provided in the HTTP request query string.
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Object} The events data in JSON format.
+ * @throws {Error} If an error occurs while retrieving the events data.
+ */
+const getAllEvents = async (req, res) => {
+  /**
+   * @type {boolean} - enabled: determines whether the rooms should be enabled or disabled.
+   * @type {number} - limit: The maximum number of rooms to retrieve.
+   * @type {string} - offset: The starting point for retrieving rooms.
+   */
+  const { enabled, limit, offset } = req.query;
+  try {
+    const start = offset || "";
+    const limitOfRooms = limit || GET_ALL_EVENTS_LIMIT_MIN;
+    const isEnabled = enabled || false;
+    const eventsData = await apiService.get(`/rooms?limit=${limitOfRooms}&enabled=${isEnabled}&start=${start}`);
+    if (eventsData.data) {
+      const propertiesToRemove = ["customer_id", "app_id", "recording_info", "template_id", "template", "customer"];
```

Comment:
you need not declare this again and again in every scope of the function just create a global constant for this and use

Reply:
Yes, thank you. Added a constant.